File: Lab 3/C++/main.cpp
#include "StringNumber.h"
using namespace Artem;

int main() 
{
    StringNumber C1;                
    StringNumber C2("12345");      
    StringNumber C3 = C2;          

    ////////////////////////////////////////////////////////////////////////
    //  "вирахувати" з об’єкта C2 символ '5';                             //
    //  "скласти"об'єкти C2 та C3 і результат "помістити" до об’єкта C1;  //                             
    ////////////////////////////////////////////////////////////////////////
    
    C2 = C2 - '5';
    C1 = C2 + C3;
    
    cout << "C1: " << C1.getValue() << endl;
    cout << "C2: " << C2.getValue() << endl; 
    cout << "C3: " << C3.getValue() << endl;

    cout << "---------------------------" << endl;
    cout << "C1: " << C1.getValue() << "  Length: " << C1.length() << endl;
    cout << "C2: " << C2.getValue() << "       Lenght: " << C2.length() << endl;
    cout << "C3: " << C3.getValue() << "      Length: " << C3.length() << endl;

    return 0;
}

File: Lab 3/C#/main.cs
using System;

namespace Artem
{
    class Program
    {
        static void Main(string[] args)
        {
            StringNumber C1 = new StringNumber();
            StringNumber C2 = new StringNumber("12345");
            StringNumber C3 = new StringNumber(C2);

            C2 = C2 - '5';
            C1 = C2 + C3;

            Console.WriteLine("C1: " + C1.GetValue());
            Console.WriteLine("C2: " + C2.GetValue());
            Console.WriteLine("C3: " + C3.GetValue());

            Console.WriteLine("---------------------------");
            Console.WriteLine($"C1: {C1.GetValue()}  Length: {C1.Length()}");
            Console.WriteLine($"C2: {C2.GetValue()}       Length: {C2.Length()}");
            Console.WriteLine($"C3: {C3.GetValue()}      Length: {C3.Length()}");
        }
    }
}
